ot3_state_manager/ot3_state_manager/messages.py

```diff
@@ -52,55 +46,94 @@ class MoveMessage(Message):
     direction: Direction
 
-    @classmethod
-    def to_message(cls, string: str) -> Optional[MoveMessage]:
-        """Parse string to Message or None."""
-        split_string = string.split(" ")
-
-        if len(split_string) == 2:
-            try:
-                axis = OT3Axis[split_string[1]]
-            except KeyError:
-                axis = None
-
-            try:
-                direction = Direction.from_symbol(split_string[0])
-            except ValueError:
-                direction = None
-
-            if axis is not None and direction is not None:
-                return cls(axis=axis, direction=direction)
+    def __eq__(self, other: Any) -> bool:
+        """Confirm that the 2 objects are equal"""
+        return isinstance(other, MoveMessage) and (
+            self.axis == other.axis and self.direction == other.direction
+        )
+
+    def __hash__(self) -> int:
+        """Return hash for instance."""
+        return hash(str(self))
+
+
+class MessageLookup:
+    """Lookup class for mapping a message to a byte value."""
+
+    MESSAGE_LOOKUP_DICT: Dict[int, Message] = {
+        0x0000: MoveMessage(OT3Axis.X, Direction.NEGATIVE),
+        0x0001: MoveMessage(OT3Axis.X, Direction.POSITIVE),
+        0x0010: MoveMessage(OT3Axis.Y, Direction.NEGATIVE),
+        0x0011: MoveMessage(OT3Axis.Y, Direction.POSITIVE),
+        0x0020: MoveMessage(OT3Axis.Z_L, Direction.NEGATIVE),
+        0x0021: MoveMessage(OT3Axis.Z_L, Direction.POSITIVE),
+        0x0030: MoveMessage(OT3Axis.Z_R, Direction.NEGATIVE),
+        0x0031: MoveMessage(OT3Axis.Z_R, Direction.POSITIVE),
+        0x0040: MoveMessage(OT3Axis.Z_G, Direction.NEGATIVE),
+        0x0041: MoveMessage(OT3Axis.Z_G, Direction.POSITIVE),
+        0x0050: MoveMessage(OT3Axis.P_L, Direction.NEGATIVE),
+        0x0051: MoveMessage(OT3Axis.P_L, Direction.POSITIVE),
+        0x0060: MoveMessage(OT3Axis.P_R, Direction.NEGATIVE),
+        0x0061: MoveMessage(OT3Axis.P_R, Direction.POSITIVE),
+        0x0070: MoveMessage(OT3Axis.Q, Direction.NEGATIVE),
+        0x0071: MoveMessage(OT3Axis.Q, Direction.POSITIVE),
+        0x0080: MoveMessage(OT3Axis.G, Direction.NEGATIVE),
+        0x0081: MoveMessage(OT3Axis.G, Direction.POSITIVE),
+        0x1000: SyncPinMessage("LOW"),
+        0x1001: SyncPinMessage("HIGH"),
+    }
+
+    REVERSE_LOOKUP_DICT: Dict[Message, int] = {
+        val: key for key, val in MESSAGE_LOOKUP_DICT.items()
+    }
+
+    FORMAT_STRING = ">H"
 
-        return None
-
-
-def parse_message(string: str) -> Union[SyncPinMessage, MoveMessage]:
-    """Parse string into a Message Type"""
-    sync_pin_message = SyncPinMessage.to_message(string)
-    move_message = MoveMessage.to_message(string)
-
-    if sync_pin_message is not None:
-        return sync_pin_message
+    @classmethod
+    def message_to_bytes(cls, message: Message) -> bytes:
+        """Looks up and returns byte value for message."""
+        if message not in cls.REVERSE_LOOKUP_DICT:
```

Comment:
do we need something like this? if it doesn't exist, you get a `KeyError` later right?

Reply:
Removed in 2cfb104355ce884c880cf4f82bae2112146bce4c